builders/xcm/fees.md

```diff
@@ -0,0 +1,230 @@
+```
+
+The total cost is `0.000005 DEV` for an XCM instruction on Moonbase Alpha.
+
+## Calculation of Fees for External Assets on Moonbeam {: #calculation-of-fees-external-asset }
+
+In the previous example with Alice, her fees were taken out of Alith's xcDOT balance. To determine the amount of a particular asset, such as xcDOT, to charge per second of local XCM execution, Moonbeam uses a concept called `UnitsPerSecond`. This concept is used by parachains to determine how much to charge in the target parachain. Nevertheless, fees can be charged in another token, for example, DOT.
+
+To find out the `UnitsPerSecond` for a given asset, you can query `assetManager.assetTypeUnitsPerSecond` and pass in the multilocation of the asset in question.
+
+If you're unsure of the multilocation, you can retrieve it using the `assetManager.assetIdType` query.
+
+For example, you can navigate to Polkadot.js Apps and under the **Developer** dropdown, choose **Chain State**. From there you can take the following steps:
+
+1. For the **selected state query** dropdown, choose **assetManager**
+2. Select the **assetIdType** extrinsic
+3. Under **Option** enter in the asset ID or toggle the **include option** off to return information for all of the assets. This example will get information for xcUNITs which has an asset ID of `42259045809535163221576417993425387648`
+4. Click the **+** button to submit the query
+
+![Get the xcUNIT asset multilocation](/images/builders/xcm/fees/fees-1.png)
+
+You can take the result of the query and then use it to query the **assetTypeUnitsPerSecond** extrinsic:
+
+1. Make sure **assetManager** is selected
+2. Select the **assetTypeUnitsPerSecond** extrinsic
+3. For **MoonbaseRuntimeXcmConfigAssetType**, choose **Xcm**
+4. Enter `1` for **parents**
+5. Select `Here` for **interior**
+6. Click the **+** button to submit the query
+
+The `UnitsPerSecond` for xcDOT is `11,285,231,116`.
+
+![Get the xcUNIT units per second value](/images/builders/xcm/fees/fees-2.png)
+
+To calculate the total cost, you'll also need the number of decimals the asset in question uses, which for xcDOT is 12 decimals. You can determine the number of decimals for any asset by [querying the asset metadata](/builders/xcm/xc20/xc20/#x-chain-assets-metadata){target=_blank}.
+
+Remember that one unit of weight is defined as one picosecond of execution time. Therefore, to get the cost of execution time for a single instruction, you can use the following formula:
+
+```
+ExecutionTime = Weight / Picosecond
+```
+
+The calculation for Moonbeam, for example, is as follows:
+
+```
+ExecutionTime = 200000000 / 10^12
+```
+
+Which means that one XCM instruction costs `0.0002` seconds of block execution time.
+
+The block exuection formula can then be used to determine how much the Alice's transfer of xcDOT to Alith costs. The formula for finding the total cost is as follows:
+
+```
+Cost = (UnitsPerSecond / Picosecond) * ExecutionTime * NumberOfInstructions
```

Comment:
You already have the Execution Time in `seconds`. You need to divide `UnitsPerSecond` by `10^decimals` to get the result in UNITS (DOT for example, and not Plank-DOT)